coursera/extractors.py

```diff
@@ -162,7 +163,12 @@ def _parse_on_demand_syllabus(self, page, reverse=False,
                     elif typename == 'programming':
                         if download_quizzes:
                             links = course.extract_links_from_programming_immediate_instructions(lecture['id'])
-
+                    
+                    elif typename == 'notebook':
+                        if download_notebooks and self._notebook_downloaded == False:
+                            logging.warning('According to notebooks platform, content will be downloaded first')
+                            links = course.extract_links_from_notebook(lecture['id'])
+                            self._notebook_downloaded = True
```

Comment:
Could you please explain why this option is necessary?

Reply:
Just curious, @moiseslodeiro, why the notebooks need to be downloaded first? Where did you get that information?

Reply:
Hi guys, I'm working on it. It is because the session Id on notebooks platform. See that the links are added to syllabus but fails after a while. This is because Coursera creates a token when access to notebook and if you don't use it in a short period of time, you need to regenerate the token (and the url also changes). I'm working on some changes and tests to commit them asap.. :-)